mask/eolearn/mask/snow_mask.py

```diff
@@ -0,0 +1,261 @@
+                     for frame in input_array]
+        return np.array(resampled)
+
+    def _adjust_cloud_mask(self, bands, cloud_mask, dem, b10):
+        """ Adjust existing cloud mask using cirrus band if L1C data and resampled red band """
+        # add to the existing cloud mask pixels found thresholding down-sampled red band and cirrus band/DEM
+        clm_b10 = np.where(b10 > self.B10_THR + self.DEM_FACTOR * dem, 1, 0) \
+            if b10 is not None else np.ones(shape=cloud_mask.shape, dtype=np.uint8)
+        return np.logical_or(
+            np.where(np.logical_and(cloud_mask == 1, self._resample_red(bands[..., 1]) > self.red_params[1]), 1, 0),
+            clm_b10).astype(np.uint8)
+
+    def _apply_first_pass(self, bands, ndsi, clm, dem, clm_temp):
+        """ Apply first pass of snow detection """
+        snow_mask_pass1 = np.where(np.logical_and(np.logical_not(clm_temp),
+                                                  np.logical_and(ndsi > self.ndsi_params[0],
+                                                                 bands[..., 1] > self.red_params[3])), 1, 0)
+
+        clm_pass1 = np.where(
+            np.logical_or(clm_temp, (bands[..., 1] > self.red_params[2]) &
+                          np.logical_not(snow_mask_pass1) & clm), 1, 0)
+
+        dem_edges = np.linspace(np.min(dem), np.max(dem),
+                                int(np.ceil((np.max(dem) - np.min(dem)) / self.dem_params[0])))
+        nbins = len(dem_edges) - 1
+        dem_hist_clear_pixels, snow_frac = None, None
+        if nbins > 0:
+            snow_frac = np.zeros(shape=(bands.shape[0], nbins))
+            dem_hist_clear_pixels = np.array([np.histogram(dem[np.logical_not(mask)], bins=dem_edges)[0]
+                                              for mask in clm_pass1])
+
+            for date, nbin in itertools.product(range(bands.shape[0]), range(nbins)):
+                if dem_hist_clear_pixels[date, nbin] > 0:
+                    dem_mask = np.logical_and(dem_edges[nbin] <= dem, dem < dem_edges[nbin + 1])
+                    in_dem_range_clear = np.where(
+                        np.logical_and(dem_mask, np.logical_not(clm_pass1[date])))
+                    snow_frac[date, nbin] = \
+                        np.sum(snow_mask_pass1[date][in_dem_range_clear]) / dem_hist_clear_pixels[date, nbin]
+        return snow_mask_pass1, snow_frac, dem_edges
+
+    def _apply_second_pass(self, bands, ndsi, dem, clm_temp, snow_mask_pass1, snow_frac, dem_edges):
+        """ Second pass of snow detection """
+        _, height, width, _ = bands.shape
+        total_snow_frac = np.sum(snow_mask_pass1, axis=(1, 2)) / (height*width)
+        snow_mask_pass2 = np.zeros(snow_mask_pass1.shape)
+        for date in range(bands.shape[0]):
+            if total_snow_frac[date] > self.ndsi_params[2]:
+                if snow_frac is not None and np.any(snow_frac[date] > self.dem_params[1]):
```

Comment:
Probably these 2 `if` sentences could be joined into a single one?

Reply:
Changed as suggested